tests/unit/test_platform_utils.py

```diff
@@ -146,22 +146,32 @@ def fake_flow():
     # Set up a database
     db_file = get_workflow_run_pub_db_path(flow_name)
     Path(db_file).parent.mkdir()
-    db_script = (
-        b"CREATE TABLE task_jobs("
-        b"cycle TEXT, name TEXT, submit_num INTEGER, platform_name TEXT);\n"
-        b"INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
-        b"    VALUES ('1', 'bar', 1, 'localhost');\n"
-        b"INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
-        b"    VALUES ('1', 'baz', 1, 'dairy');\n"
-        b"INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
-        b"    VALUES ('1', 'bar', 2, 'dairy');\n"
-        b"INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
-        b"    VALUES ('2', 'baz', 1, 'milk');\n"
-    )
-    run(
-        ['sqlite3', db_file],
-        input=db_script,
-    )
+
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+    # Deliberately add duplicate row, ensures that logic in retrieval deals
+    # with it:
+    for line in (
+        "CREATE TABLE task_jobs("
+        "cycle TEXT, name TEXT, submit_num INTEGER, platform_name TEXT);",
+        "INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
+        "    VALUES ('1', 'bar', 1, 'localhost');",
+        "INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
+        "    VALUES ('1', 'baz', 1, 'dairy');",
+        "INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
+        "    VALUES ('1', 'bar', 2, 'dairy');",
+        "INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
+        "    VALUES ('1', 'bar', 2, 'dairy');",
+        "INSERT INTO task_jobs (cycle, name, submit_num, platform_name)"
+        "    VALUES ('2', 'baz', 1, 'milk');",
+        "CREATE TABLE workflow_params(key TEXT, value TEXT);",
+        "INSERT INTO workflow_params"
+        f"    VALUES ('cylc_version', {cylc_version!r});",
+        "CREATE TABLE workflow_template_vars(key TEXT, value TEXT);",
+    ):
+        cursor.execute(line)
+    conn.commit()
+    conn.close()
```

Comment:
Instead of using an sqlite3 connection directly, could import and use
```py
with CylcWorkflowDAO(db_file, create_tables=True) as dao:
    dao.connect().execute(
```
and with that can get rid of the `CREATE TABLES` statements

Reply:
Yes, but I don't believe that it should block this.

Reply:
On the contrary, I think using the DAO-created tables means the test better reflects the ground truth of what the DB is actually like, and more likely to catch problems with DB changes